### PR TITLE
Dynamically generate variables based on release version

### PR DESCRIPTION
**Pull request by npawelek**
This is an initial commit to set intelligent default variables and default those based on version.

### CODE REVIEW

File: playbooks/maas-pre-flight.yml [hunk 1]
```diff
@@ -49,6 +49,158 @@
...
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: true
+  pre_tasks:
+    - name: Check for /etc/openstack-release
+      stat:
+        path: /etc/openstack-release
+      register: _osa_release_file
+
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: from line 62-74 I think we can wrap all that in a block making it conditional the existence of the release file.

``` yaml
- name: osa release block
  block:
    - name: Register /etc/openstack-release version
      shell: grep DISTRIB_RELEASE /etc/openstack-release | tr -dc '0-9.'
      register: _maas_product_osa_release_version

    - name: Set OSA release version fact
      set_fact:
        maas_product_osa_release_version: "{{ _maas_product_osa_release_version.stdout.split('.')[0] }}"
  when:
    - not deploy_osp | bool
    - _osa_release_file.stat.exists
```

File: playbooks/maas-pre-flight.yml [hunk 2]
```diff
@@ -49,6 +49,158 @@
...
+        maas_product_osa_release_version: "{{ _maas_product_osa_release_version.stdout.split('.')[0] }}"
+      when:
+        - _maas_product_osa_release_version.changed | bool
+
+    - name: Check for /etc/rpc-release
+      stat:
+        path: /etc/rpc-release
+      register: _rpc_release_file
+
+    - name: Register /etc/rpc-release version
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: same for 81-86

File: playbooks/maas-pre-flight.yml [hunk 3]
```diff
@@ -49,6 +49,158 @@
...
+        maas_product_rpco_release_version: "{{ _maas_product_rpco_release_version.stdout.split('.')[0] }}"
+      when:
+        - _maas_product_rpco_release_version.changed | bool
+
+    - name: Check for /etc/rhosp-release
+      stat:
+        path: /etc/rhosp-release
+      register: _osp_release_file
+
+    - name: Register /etc/rhosp-release version
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: same for 99-110

File: playbooks/maas-pre-flight.yml [hunk 4]
```diff
@@ -49,6 +49,158 @@
...
+  tasks:
+    - name: Set RPCO Kilo compatibility variables
+      set_fact:
+        openstack_upstream_domain: "rpc-repo.rackspace.com"
+        repo_pip_default_index: "http://{‌{ openstack_upstream_domain }‌}/pools"
+      when:
+        - maas_product_rpco_release_version is defined
+        - maas_product_osa_release_version == 11
+
+    - name: Set legacy swift_recon_path
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: maybe this can be moved into the osa bock?

File: playbooks/maas-pre-flight.yml [hunk 5]
```diff
@@ -49,6 +49,158 @@
...
+        - _osp_release_file.stat.exists
+
+    - name: Set OSP release version fact
+      set_fact:
+        maas_product_osp_release_version: "{{ _maas_product_osp_release_version.stdout.split('.')[0] }}"
+      when:
+        - _maas_product_osp_release_version.changed | bool
+
+  tasks:
+    - name: Set RPCO Kilo compatibility variables
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: this might be able to be moved into the rpco-block

File: playbooks/maas-pre-flight.yml [hunk 6]
```diff
@@ -49,6 +49,158 @@
...
+    - name: Set legacy swift_recon_path
+      set_fact:
+        swift_recon_path: "/usr/local/bin"
+      when:
+        - maas_product_osa_release_version == 11
+        - group['swift_all'] is defined
+
+    # TODO(npawelek): if swift is found, check for maas_swift_accesscheck_password
+
+    - name: Set RPCO Liberty compatibility variables
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: maybe this can be moved into the rpco block

File: playbooks/maas-pre-flight.yml [hunk 7]
```diff
@@ -49,6 +49,158 @@
...
+      set_fact:
+        repo_server_port: 8181
+        openstack_repo_url: "http://{‌{ internal_lb_vip_address }‌}:{‌{ repo_server_port }‌}"
+        openstack_upstream_domain: "rpc-repo.rackspace.com"
+        repo_pip_default_index: "http://{‌{ openstack_upstream_domain }‌}/pools"
+      when:
+        - maas_product_rpco_release_version is defined
+        - maas_product_osa_release_version == 12
+
+    - name: Set RPCO rpc_release compatibility variable
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: same as above

File: playbooks/maas-pre-flight.yml [hunk 8]
```diff
@@ -49,6 +49,158 @@
...
+        - maas_product_osa_release_version <= 13
+
+    - name: Set RPCO holland variables
+      set_fact:
+        maas_holland_enabled: true
+      when:
+        - maas_product_rpco_release_version is defined
+        - maas_product_osa_release_version <= 12
+
+    - name: Set RPCO holland variables
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: same as above

File: playbooks/maas-pre-flight.yml [hunk 9]
```diff
@@ -49,6 +49,158 @@
...
+    # TODO(npawelek): handle default _maas_excluded_checks and overridden
+    # maas_excluded_checks and perform a union
+
+    # TODO(npawelek): handle default _maas_excluded_alarms and overridden
+    # maas_excluded_alarms and perform a union
+
+    # TODO(npawelek): handle default _maas_excluded_devices and overridden
+    # maas_excluded_devices and perform a union
+
+    - name: Set OSA product metadata fact
```
Comment (cloudnull) [inline playbooks/maas-pre-flight.yml]: same as above

**Comment by npawelek** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by npawelek** [comment]
recheck_xenial_queens_deploy

**Comment by npawelek** [comment]
recheck_trusty_liberty_deploy

**Comment by npawelek** [comment]
recheck_bionic_mnaio_osp13_deploy

**Pull request merged**